### DISCUSSION TITLE
[C.Deliv] Confidence Labels

### DISCUSSION THREAD

**Discussion by erkist** [cdfoundation/sig-events#37] [General]
In PR https://github.com/cdfoundation/sig-events/pull/23 the concept of a Confidence Label is proposed, with a corresponding event, described as follows:

> Announces that a Confidence Label Verdict (a.k.a. Confidence Level) for one or more source changes, artifacts, component versions or compositions has been set or updated.
> 
> Confidence Label Verdict Events MUST include the following attributes:
> 
> - **Confidence Label Verdict ID**: Unique identifier of confidence label verdict.
> - **Name**: Machine-friendly and human-friendly name of confidence label
> - **Verdict**: The verdict of this confidence label result (Success, Failure, Inconclusive)
> - **Targets**: List of sources, artifacts, component versions and compositions this confidence label verdict applies to
> 
> Optional attributes:
> 
> - **Causes**: List of e.g. pipeline runs or test suites that lead to this confidence label verdict.
> - **Sub Confidence Label Verdicts**: List of other confidence label verdicts that resulted in this confidence label verdict. Used for “summarizing” (e.g. “parent”) confidence labels.

Related concepts:
* [ConfidenceLevelModified in Eiffel](https://github.com/eiffel-community/eiffel/blob/master/eiffel-vocabulary/EiffelConfidenceLevelModifiedEvent.md)
* [Confidence Label in ESCI](https://github.com/erkist/sig-interoperability/tree/erkist-esci/workstreams/events_in_cicd/input/enterprise-scale-ci#confidence-label)
* [Badges at eBay](https://hackmd.io/__6IAnVuR8C8mPD--XMbWg?view#eBay)

Do we want to represent this concept in the spec? If so, how? What events do we want to have attached to it?

**Comment by m-linner-ericsson** [comment]
I think the spec would be missing quite much if not able to address this concept. Adding the confidence in the "Artifact Published" event would need the event to be resent every time you need to update your confidence about it. Thus I would vote for adding a separate event.
